docs/tutorials/metadata-logging.rst

```diff
@@ -0,0 +1,137 @@
+.. _metadata-logging-tutorial:
+
+############################
+ Arbitrary Metadata Logging
+############################
+
+This tutorial demonstrates how to use Arbitrary Metadata Logging in Determined AI to log custom metadata for your experiments.
+
+**Why Use Arbitrary Metadata Logging?**
+
+Arbitrary Metadata Logging allows you to:
+
+- Capture experiment-specific information beyond standard metrics
+- Compare and analyze custom data across experiments
+- Filter and sort experiments based on custom metadata
+
+******************
+ Logging Metadata
+******************
+
+You can log metadata using the Determined Core API. Here's how to do it in your training code:
+
+1. Import the necessary module:
+
+   .. code:: python
+
+      from determined.core import Context
+
+2. In your trial class, add a method to log metadata:
+
+   .. code:: python
+
+      def log_metadata(self, context: Context):
+          context.train.report_metadata({
+              "dataset_version": "MNIST-v1.0",
+              "preprocessing": "normalization",
+              "hardware": {
+                  "gpu": "NVIDIA A100",
+                  "cpu": "Intel Xeon"
+              }
+          })
+
+3. Call this method in your training loop:
+
+   .. code:: python
+
+      def train_batch(self, batch: TorchData, epoch_idx: int, batch_idx: int):
+          # Existing training code...
+          
+          if batch_idx == 0:
+              self.log_metadata(self.context)
+
+          # Rest of the training code...
+
+This example logs metadata at the beginning of each epoch. Adjust the frequency based on your needs.
+
+*******************************
+ Viewing Metadata in the WebUI
+*******************************
+
+To view logged metadata:
+
+1. Open the WebUI and navigate to your experiment.
+2. Click on the trial you want to inspect.
+3. In the trial details page, find the "Metadata" section under the "Overview" tab.
+
+***********************************
+ Filtering and Sorting by Metadata
+***********************************
+
+The :ref:`Web UI <web-ui-if>` allows you to filter and sort experiments based on logged metadata:
+
+1. Navigate to the Experiments List page in the WebUI.
+2. Click on the filter icon.
+3. Select a metadata field from the dropdown menu.
+4. Choose a condition (is, is not, or contains) and enter a value.
+5. Click "Apply" to filter the experiments based on the metadata.
+
+For more detailed instructions on filtering and sorting, refer to the WebUI guide:
+
+Performance Considerations
+==========================
+
+When using Arbitrary Metadata Logging, consider the following:
+
+- Metadata is stored efficiently for fast retrieval and filtering.
+- Avoid logging very large metadata objects, as this may impact performance.
+- Use consistent naming conventions for keys to make filtering and sorting easier.
+- For deeply nested JSON structures, filtering and sorting are supported at the top level.
+
+Example Use Case
+================
```

Comment:
@keita-determined @AmanuelAaron this is placeholder text and the code shown is just for placeholder purposes. Do we need an example use case? If so, what are the proper steps?

Reply:
i'll use the PostRunMetadata example from the demo